enhancements/installer/connected-assisted-installer.md

```diff
@@ -0,0 +1,324 @@
+create private VLANs, configure DHCP/PXE servers, or manage other
+aspects of the infrastructure surrounding the hardware where the
+cluster will run. If they do have the required privileges, they may
+not want to delegate them to the OpenShift installer for an IPI
+installation, preferring instead to use their existing tools and
+processes for some or all of that configuration. They are willing to
+accept that the cluster they build may not have all of the same
+features as a cluster with installer-provisioned infrastructure
+on day 1, but that by taking additional
+steps later they will be able to add more automatic infrastructure
+management and make it fully production-ready.
+
+### Goals
+
+- Make initial deployment of usable and supportable clusters simpler.
+- Move more infrastructure configuration from day 1 to day 2.
+- Support connected on-premise deployments.
+- Support existing IPI features, especially for day 2 cluster
+  management and scale-out.
+
+### Non-Goals
+
+- Because the initial focus is on bare metal, this enhancement does
+  not exhaustively cover variations needed to offer similar features
+  on other platforms (such as changes to image formats, the way a host
+  boots, etc.). It is desirable to support those platforms, but that
+  work will be described separately.
+- Environments with restricted networks where hosts cannot reach the internet unimpeded
+  ("disconnected" or "air-gapped") will require more work to support
+  this installation workflow than simply packaging the hosted solution
+  built to support fully connected environments. The work to support
+  disconnected environments will be covered by a future enhancement.
+- Replace the existing OpenShift installer.
+- Describe how these workflows would work for multi-cluster
+  deployments managed with Hive or ACM.
+
+## Proposal
+
+There are several separate changes to enable the assisted installer
+workflows, including a GUI front-end for the installer, a cloud-based
+orchestration service, and changes to the installer and bootstrapping
+process.
+
+The process starts when the user goes to an "assisted installer"
+application running on `cloud.redhat.com`, enters details needed by
+the installer (OpenShift version, ssh keys, proxy settings, etc.), and
+then downloads a live RHCOS ISO image with the software and settings
+they need to complete the installation locally.
+
+The user then boots the live ISO on each host they want to be part of
+the cluster (control plane and workers). They can do this by hand
+using thumb drives, by attaching the ISO using virtual media support
+in the BMC of the host, or any other way they choose.
+
+When the ISO boots, it starts an agent that communicates with the REST
+API for the assisted installer service running on `cloud.redhat.com`
+to receive instructions. The agent registers the host with the
+service, using the user's pull secret embedded in the ISO's ignition config
```

Comment:
@romfreiman do we plan to embed the user pull secret to the ISO?
Currently this isn't the case and I think we can use some other token (that the cloud service generate) for the authentication rather than the user pull secret. 
In general I think it's preferable not to have user secrets / passwords in the ISO.

Reply:
If we have concerns about holding user secrets server-side, this could be the service encrypting the pull secret to itself, storing the encrypted blob in the ISO, and just hanging on to the encryption key.